specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2022-11-30-preview/definitions/MigrateMySqlAzureDbForMySqlSyncTask.json

```diff
@@ -0,0 +1,478 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Database Migration Service Resource Provider",
+    "version": "2022-11-30-preview"
+  },
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {},
+  "definitions": {
+    "MigrateMySqlAzureDbForMySqlSyncTaskInput": {
+      "type": "object",
+      "description": "Input for the task that migrates MySQL databases to Azure Database for MySQL for online migrations",
+      "properties": {
+        "sourceConnectionInfo": {
+          "description": "Connection information for source MySQL",
+          "$ref": "./Common.json#/definitions/MySqlConnectionInfo"
+        },
+        "targetConnectionInfo": {
+          "description": "Connection information for target Azure Database for MySQL",
+          "$ref": "./Common.json#/definitions/MySqlConnectionInfo"
+        },
+        "selectedDatabases": {
+          "type": "array",
+          "description": "Databases to migrate",
+          "items": {
+            "$ref": "#/definitions/MigrateMySqlAzureDbForMySqlSyncDatabaseInput"
+          }
+        },
+        "startedOn": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Parameter to specify when the migration started"
+        },
+        "sourceServerResourceId": {
+          "type": "string",
+          "description": "Optional resource Id of the source server if it is an azure instance"
+        },
+        "targetServerResourceId": {
+          "type": "string",
+          "description": "Optional resource Id of the target server"
+        },
+        "optionalAgentSettings": {
+          "type": "object",
+          "description": "Optional parameters for fine tuning the data migration",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

Please avoid additionalProperties pattern unless it can't reasonably be schematized.

Reply:
We can probably schematize these ones.